LambdaFunction/index.js

```diff
@@ -1,171 +1,272 @@
+            this.response.cardRenderer(SKILL_NAME, result);
+            this.response.speak(speechOutput);
+            this.emit(':responseReady');
+        });
     },
-    handle(handlerInput) {
-        const intentName = handlerInput.requestEnvelope.request.intent.name;
-        const speechText = `You just triggered ${intentName}`;
-
-        return handlerInput.responseBuilder
-            .speak(speechText)
-            //.reprompt('add a reprompt if you want to keep the session open for the user to respond')
-            .getResponse();
-    }
-};
+    'AMAZON.HelpIntent': function () {
+        const speechOutput = HELP_MESSAGE;
+        const reprompt = HELP_REPROMPT;
 
-// Generic error handling to capture any syntax or routing errors. If you receive an error
-// stating the request handler chain is not found, you have not implemented a handler for
-// the intent being invoked or included it in the skill builder below.
-const ErrorHandler = {
-    canHandle() {
-        return true;
+        this.response.speak(speechOutput).listen(reprompt);
+        this.emit(':responseReady');
+    },
+    'AMAZON.CancelIntent': function () {
+        this.response.speak(STOP_MESSAGE);
+        this.emit(':responseReady');
+    },
+    'AMAZON.StopIntent': function () {
+        this.response.speak(STOP_MESSAGE);
+        this.emit(':responseReady');
     },
-    handle(handlerInput, error) {
-        console.log(`~~~~ Error handled: ${error.message}`);
-        const speechText = `Sorry, I couldn't understand what you said. Please try again.`;
-
-        return handlerInput.responseBuilder
-            .speak(speechText)
-            .reprompt(speechText)
-            .getResponse();
-    }
 };
 
-function httpGet() {
-    return new Promise(((resolve, reject) => {
-        var options = {
-            host: 'api.icndb.com',
-            port: 443,
-            path: '/jokes/random',
-            method: 'GET',
-        };
-
-        const request = https.request(options, (response) => {
-            response.setEncoding('utf8');
-            let returnData = '';
-
-            response.on('data', (chunk) => {
-                returnData += chunk;
-            });
-
-            response.on('end', () => {
-                resolve(JSON.parse(returnData));
-            });
-
-            response.on('error', (error) => {
-                reject(error);
-            });
+function httpsGet(apiPath, callback) {
+
+    var options = {
+        port: 443,
+        host: 'softwidgetapi.herokuapp.com',
+        path: apiPath,
+        method: 'GET',
+    };
+
+    var req = https.request(options, res => {
+        res.setEncoding('utf8');
+        var responseString = "";
+
+        //accept incoming data asynchronously
+        res.on('data', chunk => {
+            responseString = responseString + chunk;
+        });
+
+        //return the data when streaming is complete
+        res.on('end', () => {
+            console.log(responseString);
```

Comment:
Remove console

Reply:
Can we skip this or, can we ask Mohit to do this when he completes his work?